warehouse/query-core/src/main/java/datawave/query/jexl/functions/QueryFunctionsDescriptor.java

```diff
@@ -43,24 +48,44 @@ public QueryJexlArgumentDescriptor(ASTFunctionNode node, String namespace, Strin
         
         @Override
         public JexlNode getIndexQuery(ShardQueryConfiguration config, MetadataHelper helper, DateIndexHelper dateIndexHelper, Set<String> datatypeFilter) {
-            // return the true node if unable to parse arguments.
-            JexlNode returnNode = TRUE_NODE;
-            
-            if (name.equals("between")) {
-                
-                JexlNode geNode = JexlNodeFactory.buildNode(new ASTGENode(ParserTreeConstants.JJTGENODE), args.get(0), args.get(1).image);
-                JexlNode leNode = JexlNodeFactory.buildNode(new ASTLENode(ParserTreeConstants.JJTLENODE), args.get(0), args.get(2).image);
-                
-                // now link em up
-                
-                returnNode = BoundedRange.create(JexlNodeFactory.createAndNode(Arrays.asList(geNode, leNode)));
-            } else if (name.equals("length")) {
-                // create a regex node with the appropriate number of matching characters
-                
-                returnNode = JexlNodeFactory.buildNode(new ASTERNode(ParserTreeConstants.JJTERNODE), args.get(0), ".{" + args.get(1).image + ','
-                                + args.get(2).image + '}');
+            switch (name) {
+                case BETWEEN:
+                    JexlNode geNode = JexlNodeFactory.buildNode(new ASTGENode(ParserTreeConstants.JJTGENODE), args.get(0), args.get(1).image);
+                    JexlNode leNode = JexlNodeFactory.buildNode(new ASTLENode(ParserTreeConstants.JJTLENODE), args.get(0), args.get(2).image);
+                    // Return a bounded range.
+                    return BoundedRange.create(JexlNodeFactory.createAndNode(Arrays.asList(geNode, leNode)));
+                case LENGTH:
+                    // Return a regex node with the appropriate number of matching characters
+                    return JexlNodeFactory.buildNode(new ASTERNode(ParserTreeConstants.JJTERNODE), args.get(0), ".{" + args.get(1).image + ','
+                                    + args.get(2).image + '}');
+                case QueryFunctions.MATCH_REGEX:
+                    // Return an index query.
+                    return getIndexQuery();
+                default:
+                    // Return the true node if unable to parse arguments.
+                    return TRUE_NODE;
+            }
+        }
+        
+        private JexlNode getIndexQuery() {
+            JexlNode node0 = args.get(0);
+            final String value = args.get(1).image;
+            if (node0 instanceof ASTIdentifier) {
+                final String field = JexlASTHelper.deconstructIdentifier(node0.image);
+                return JexlNodeFactory.buildNode((ASTEQNode) null, field, value);
```

Comment:
might also need to tweak the planner since it looks like we try to push down regular expressions before attempting to expand against the index which obviously won't work for index-only fields

Reply:
Glad to hear it.

Reply:
I would still like a more end-to-end test though, please.

Reply:
Will do, currently writing one up.